python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -452,6 +472,46 @@ def set_prometheus_updater_data(self):
                 num_recovering += 1
         self.prom_metrics.recovering_nodes.set(num_recovering)
 
+    def _report_pending_infeasible(self, unfulfilled: List[ResourceDict]):
+        """Emit event messages for infeasible or unschedulable tasks.
+
+        This adds messages to the event summarizer for warning on infeasible
+        or "cluster full" resource requests.
+
+        Args:
+            unfulfilled: List of resource demands that would be unfulfilled
+                even after full scale-up.
+        """
+        pending = []
+        infeasible = []
+        for bundle in unfulfilled:
+            placement_group = any("_group_" in k for k in bundle)
+            if placement_group:
+                continue
+            if self.resource_demand_scheduler.is_feasible(bundle):
+                pending.append(bundle)
+            else:
+                infeasible.append(bundle)
+        if pending:
+            if self.load_metrics.cluster_full_reported:
```

Comment:
What could it mean if there's a pending bundle but cluster full isn't reported? 
This could happen if the Ray version is old. 
Is it possible even if the Ray version matches the autoscaler version, and is there anything that should be logged in that case?

Reply:
It just means tasks are pending, nothing to warn about in that case. Maybe I will rename this to cluster full of actors.

Reply:
Thanks, makes more sense now!

Reply:
Hmm, though you can still get into resource deadlock situations without actors -- say everything is clogged with tasks that depend on pending tasks. That's more of an edge case, of course.